File: algorithm and problem solving/lecture/towerofhanoi.c
#include <stdio.h>

void towerOfHanoi(int n, char source, char aux, char dest) {
    if (n == 1) {
        printf("Move disk 1 from %c to %c\n", source, dest);
        printf("we ");
        return;
    }
    towerOfHanoi(n - 1, source, dest, aux);
    printf("Move disk %d from %c to %c\n", n, source, dest);
    printf("  ----   \n");
    towerOfHanoi(n - 1, aux, source, dest);
}

int main() {
    int n = 4; // Number of disks
    towerOfHanoi(n, 'A', 'B', 'C');
    return 0;
}

File: Algorithm And Program Solving/lecture/towerofhanoi.py
def tower_of_hanoi(n, source, destination, spare, ):
    if n == 1:
        print(f"Move disk 1 from {source} to {destination}")
        return 
    
    tower_of_hanoi(n - 1, source, spare, destination, ) # this function makes space for the big disk at destination
    print(f"Move disk {n} from {source} to {destination}") # this function moved the big disk to destination
    tower_of_hanoi(n - 1, spare, destination, source, ) # this function moves the small disks to destination

# Example usage
tower_of_hanoi(4, 'A', 'C', 'B')